docs/src/pages/guides/composition/composition.md

```diff
@@ -96,3 +99,17 @@ class ListItemLink extends React.Component {
 Here is a demo with [React Router](https://github.com/ReactTraining/react-router):
 
 {{"demo": "pages/guides/composition/ComponentProperty.js"}}
+
+### Caveat with refs
+Some components such as `ButtonBase` (and therefore `Button`) require access to the 
+underlying DOM node. This was previously done with `ReactDOM.findDOMNode(this)`.
+However `findDOMNode` was deprecated (which disqualifies its usage in react's concurrent mode)
```

Comment:
*react's concurrent mode -> react's strict mode?*

There are very few references to "concurrent mode" in the official documentation. As far as I know, the depreciation is about allowing the React team to simplify the core implementation and solving the children fragment WFT?! moment.

Reply:
So this is my issue with the `findDOMNode` deprecation: If it were only for strict mode it wouldn't really matter that much at the moment. After all "it's just a warning". The only real implication is that reacts concurrent mode (which was previously named async mode and enables e.g. Time Slicing) requires strict mode compliance. So it has actual implications for this mode. Strict mode is not doing anything in production but concurrent mode actually does. That's what I wanted to emphasize here i.e. we're not just introducing a breaking change to be pretty but to support upcoming react features.

Reply:
Ok great 👍

Reply:
If this was targeted at master I would agree since it's alpha only at the moment (although we couldn't hold back with hooks too ;) ) but since this is targeted at `next` and I guess we aim at 16.7 strict mode compliance with v4 this is fine IMO.